weave

```diff
@@ -205,6 +205,40 @@ dns_arg_count() {
     fi
 }
 
+collect_rewrite_hosts_args() {
+    REWRITE_HOSTS_ARG_COUNT=0
+    case "$1" in
+    --rewrite-hosts)
+        REWRITE_HOSTS=1
+        REWRITE_HOSTS_ARG_COUNT=1
+        ;;
```

Comment:
I disagree on both points.

This adds a fair bit of complication to the code in this case. Please get rid of it.